### PR TITLE
Music josh

### PR DESCRIPTION
**Pull request by JoshCTFS**
Music + Rearrangement of Prefabs + Re-editing of scores

SoundManager script created to manage the background music.
Sound also added to the death of the Light and Medium enemies as well as scores edited once again.

Currently experiencing issue when testing where enemies disappear, should not be occuring based on script and requires testing.

SoundManager used to switch audio playing based on the current Scene.
Still need to find how to add another audio file to the component other than the array, in order to utilise the starting sound for when the Game scene loads.

(May or may not have the wrong starting music playing :^) )

### CODE REVIEW

File: Assets/_Scripts/Managers/SoundManager.cs [hunk 1]
```diff
@@ -1,31 +1,52 @@
...
-    }
-    // Start is called before the first frame update
-    void Start()
-    {
-        
+        if (FindObjectsOfType<SoundManager>().Length > 1) //Destroys other Objects with the SoundManager Script if more than 1 is present
+        {
+            Destroy(gameObject);
+            return;
+        }
```
Comment (SolomanDavis) [inline Assets/_Scripts/Managers/SoundManager.cs]: This class should just be a `Singleton<SoundManager>` instead and be placed in the `Title` scene

File: Assets/_Scripts/Managers/SoundManager.cs [hunk 2]
```diff
@@ -1,31 +1,52 @@
...
+                audioSource = gameObject.AddComponent<AudioSource>();
+            }
+
+        audioSource.loop = true; //Used to loop the audio
+        SceneManager.sceneLoaded += OnSceneLoaded; 
     }
 
-    // Update is called once per frame
-    void Update()
+    private void OnSceneLoaded(Scene scene, LoadSceneMode mode)
```
Comment (SolomanDavis) [inline Assets/_Scripts/Managers/SoundManager.cs:19]: 👍

File: Assets/_Scripts/Managers/SoundManager.cs [hunk 3]
```diff
@@ -1,31 +1,52 @@
...
+            audioSource.clip = backgroundMusic[sceneIndex]; //Used to ensure that the corresponding track plays
+            audioSource.Play();
+        }
+        else
+        {
+            Debug.LogWarning("No music assigned for scene index: " + sceneIndex);
+        }
 
+        if (scene.name == "Game") //Unused but want to play a certain sound upon the scene loading. Need to find way to designate another audio clip within same component.
+        {
```
Comment (SolomanDavis) [inline Assets/_Scripts/Managers/SoundManager.cs:34]: We can explore this together when we go to build our final build artifact

**Review by SolomanDavis — CHANGES_REQUESTED** [review]

**Comment by JoshCTFS** [comment]
Music Implemented (Audio Object destroyed on Game scene load)

**Review by SolomanDavis — APPROVED** [review]
Let's merge into master for now to avoid the bigger merge conflicts

**Pull request merged**